chips/sifive/src/uart.rs

```diff
@@ -91,15 +93,21 @@ pub struct Uart<'a> {
     rx_len: Cell<usize>,
     rx_position: Cell<usize>,
     rx_status: Cell<UARTStateRX>,
+
+    deferred_call: &'a DeferredCall<T>,
 }
 
 #[derive(Copy, Clone)]
 pub struct UartParams {
     pub baud_rate: u32,
 }
 
-impl<'a> Uart<'a> {
-    pub fn new(base: StaticRef<UartRegisters>, clock_frequency: u32) -> Uart<'a> {
+impl<'a, T> Uart<'a, T> {
+    pub fn new(
+        base: StaticRef<UartRegisters>,
+        clock_frequency: u32,
+        deferred_call: &'a DeferredCall<T>,
```

Comment:
You should be able to use a `DynamicDeferredCall` here instead. Have a look at other examples in the `chips` directory.

Reply:
`DynamicDeferredCall` has notable size overhead relative to this implementation. I suspect switching to it would cost ~250 bytes on boards with two UARTs

Reply:
Ah ok